packages/gatsby/src/commands/build.ts

```diff
@@ -76,15 +80,9 @@ module.exports = async function build(program: IBuildArgs): Promise<void> {
     graphqlTracing: program.graphqlTracing,
   })
 
-  const {
-    processPageQueries,
-    processStaticQueries,
-  } = queryUtil.getInitialQueryProcessors({
-    parentSpan: buildSpan,
-    graphqlRunner,
-  })
+  const { queryIds } = await calculateDirtyQueries({ store })
 
-  await processStaticQueries()
+  await runStaticQueries({ queryIds, parentSpan: buildSpan, store })
```

Comment:
We probably would want to pass `graphqlRunner` instance here (and in `runPageQueries`) to avoid lazily creating multiple ones